### DISCUSSION TITLE
problems with compiling EWW

### DISCUSSION THREAD

**Discussion by Vodahlyest** [elkowar/eww#829] [Q&A] [answered]
Greetings! i am in search of help with installing EWW in Endeavour OS. I've noticed that this widget system is very popular these days, and I've decided to try it with my fresh Endeavour OS install with i3 WM, but when i try to compile it form git, i get an error message with following text: "warning: associated function `validate` is never used" (screenshot is pinned). Hope someone will help me with solving that problem
![2023-06-29-18:24:41-screenshot](https://github.com/elkowar/eww/assets/104577497/6afd9e62-d7b7-4d4b-8ade-2ecc98a86e94)

**Comment by viandoxdev** [comment, marked as answer]
Well looks like your build worked. The terminal color scheme you're using doesn't make it obvious, but the line
```
Finished release [optimized build] target(s) in 0.44s 
```
means the build succeeded. The whole `validate` stuff is just a warning you can ignore. However what did fail is your first build, I can't see the command used above, but this one failed. The error that caused it has already been brought up over at #810 there is a PR up at #815 that should fix this, however since the x11 build seemed to finish you shouldn't have to.

**Reply by Vodahlyest** [reply]
Thank you, brother! when i read your reply and tried to run eww, i stuck with the message, like, "the eww does not exist" or something like that. After that, i really wanted to ask you about how to run it, but soon enough i realised that i haven't even read documentation, and after reading it, i figured out how to run it.

so, thank you again with help me! :)
